lib/DiscoveryStreamFeed.jsm

```diff
@@ -878,6 +917,7 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
                 spocs: newSpocs,
               },
             }));
+            this._handleSpocsFill();
```

Comment:
Similar to above, do we need to send this update at all?

Reply:
Ah I think I see how you're seeing the recalc thing.

You send the ping in four cases.

1. When the server returns fresh spocs.
2. When an impression updates the spocs.
3. When the page renders fresh.
4. When the page re renders and is not fresh.

Does that add up to you?

Reply:
Based on the specs of this ping, yes. Analytics wanted to capture all the filtered SPOCS by frequency capping. Over here, a `SPOC_IMPRESSION` calls the `frequencyCapSpocs` which may get a SPOC filtered.